plugins/woocommerce/client/legacy/js/frontend/single-product.js

```diff
@@ -326,14 +330,79 @@ jQuery( function( $ ) {
 				}
 				captionEl.children[0].textContent = item.title;
 				return true;
-			}
+			},
+			timeToIdle: 0, // Ensure the gallery controls are always visible to avoid keyboard navigation issues.
 		}, wc_single_product_params.photoswipe_options );
 
 		// Initializes and opens PhotoSwipe.
 		var photoswipe = new PhotoSwipe( pswpElement, PhotoSwipeUI_Default, items, options );
+
+		photoswipe.listen( 'afterInit', function() {
+			self.trapFocusPhotoswipe( true );
+		});
+
+		photoswipe.listen( 'close', function() {
+			self.trapFocusPhotoswipe( false );
+		});
+
+		photoswipe.listen( 'destroy', function() {
+			var focusCurrentTargetId = setTimeout( function() {
+				currentTarget.focus();
+				clearTimeout( focusCurrentTargetId );
+			}, 500 );
+		});
```

Comment:
Is there any case where `close` and `destroy` might be triggered separately? I wonder if those two events could be merged.

Reply:
@Aljullu The callback fired by the `close` event removes the event listener from the Photoswipe container, therefore, the container needs to exist in the DOM. On the other hand, the `destroy` event goes off when the container is completely destroyed, so no more DOM manipulation will be done by the Photoswipe script, and it's safe to focus on the button that had opened the gallery before.

Reply:
Thanks! I answered in the thread above as I think they ended up being related. :slightly_smiling_face: